### PR TITLE
Scanese lidar

### PR DESCRIPTION
**Pull request by JasonGibson274**
Updated the repository to deal with the new lidar publishing directly to pointcloud while gazebo publishes laser_geometry still. Made the lidar 360 degrees in simulation it still has some buggy results inn simulation.

### CODE REVIEW

File: gazebo/igvc_control/CMakeLists.txt [hunk 1]
```diff
@@ -41,4 +43,7 @@ catkin_install_python(
 install(PROGRAMS
         scripts/spawn_jaymii.sh
         scripts/controller_spawner.py
-DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
\ No newline at end of file
+DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
+
+add_subdirectory(nodes/control)
+add_subdirectory(nodes/scan_to_pointcloud)
```
Comment (AaronMcDaniel) [inline gazebo/igvc_control/CMakeLists.txt]: Nope

File: gazebo/igvc_control/launch/gazebo.launch [hunk 2]
```diff
@@ -44,11 +44,11 @@
     <!-- needed for rviz -->
     <param name="robot_description" command="cat $(find igvc_description)/urdf/Jaymii.urdf" />
     <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
-    
+
     <!-- convert laser scan to point cloud for LIDAR -->
     <node name="scan_to_pointcloud" pkg="igvc" type="scan_to_pointcloud" output="screen" />
```
Comment (atuttle7) [inline gazebo/igvc_control/launch/gazebo.launch]: Moved to IGVC control.

File: gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp [hunk 3]
```diff
@@ -16,17 +16,17 @@ void scanCallback(const sensor_msgs::LaserScan::ConstPtr& msg)
     return !std::isnan(range) && range > scanData.range_min && range < scanData.range_max;
   };
 
-  for (unsigned int i = 0; i < scanData.ranges.size(); i++)
+    for (unsigned int i = 0; i < scanData.ranges.size(); i++)
```
Comment (JasonGibson274) [inline gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp]: fix spacing

File: gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp [hunk 4]
```diff
@@ -16,17 +16,17 @@ void scanCallback(const sensor_msgs::LaserScan::ConstPtr& msg)
...
   {
     float& range = scanData.ranges[i];
     if (range > scanData.range_max || range < scanData.range_min)
       continue;
     // Too close
-    if (range < 0.5)
+    if (range < 0.1)
```
Comment (AaronMcDaniel) [inline gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp]: Make a parameter

File: gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp [hunk 5]
```diff
@@ -16,17 +16,17 @@ void scanCallback(const sensor_msgs::LaserScan::ConstPtr& msg)
...
   };
 
-  for (unsigned int i = 0; i < scanData.ranges.size(); i++)
+    for (unsigned int i = 0; i < scanData.ranges.size(); i++)
   {
     float& range = scanData.ranges[i];
     if (range > scanData.range_max || range < scanData.range_min)
       continue;
     // Too close
-    if (range < 0.5)
```
Comment (atuttle7) [inline gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp]: No magic numbers. Make parameters through ros params

File: gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp [hunk 6]
```diff
@@ -16,17 +16,17 @@ void scanCallback(const sensor_msgs::LaserScan::ConstPtr& msg)
...
+    if (range < 0.1)
       range = scanData.range_max + 1;
     // Too far
-    else if (range > 6.0)
-      range = scanData.range_max + 1;
+    /*else if (range > 6.0)
+      range = scanData.range_max + 1;*/
```
Comment (AaronMcDaniel) [inline gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp]: Remove this comment

File: gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp [hunk 7]
```diff
@@ -16,17 +16,17 @@ void scanCallback(const sensor_msgs::LaserScan::ConstPtr& msg)
...
     if (range > scanData.range_max || range < scanData.range_min)
       continue;
     // Too close
-    if (range < 0.5)
+    if (range < 0.1)
       range = scanData.range_max + 1;
     // Too far
-    else if (range > 6.0)
-      range = scanData.range_max + 1;
+    /*else if (range > 6.0)
```
Comment (WinfinitytheStar) [inline gazebo/igvc_control/nodes/scan_to_pointcloud/main.cpp]: Remove the "commented out section"

File: igvc/src/global_mapper/main.cpp [hunk 8]
```diff
@@ -45,7 +45,7 @@ void icp_transform(pcl::PointCloud<pcl::PointXYZ>::Ptr input)
         new_points.push_back(searchPoint);
       }
     }
-    new_points.header.frame_id = "/odom";
+    new_points.header.frame_id = "/lidar";
```
Comment (JasonGibson274) [inline igvc/src/global_mapper/main.cpp]: should be odom

File: igvc/src/filter_lidar/main.cpp [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+ros::Publisher _pointcloud_pub;
+
+double x_offset, y_offset, y_size, x_size;
+
+void point_cloud_callback(const pcl::PointCloud<pcl::PointXYZ>::ConstPtr& msg)
+{
+  pcl::PointCloud<pcl::PointXYZ> result;
+  for (auto it = msg->points.begin(); it != msg->points.end(); it++)
+  {
+    if (!((it->x + x_offset < x_size && it->x + x_offset > -x_size) &&
```
Comment (jgkamat) [inline igvc/src/filter_lidar/main.cpp]: IMO this should be a function for figuring out if something is inside a rect (do you have a geometry library?), but you coul also do

`abs(it->x + x_offset < x_size)` as well to make this a bit more readable.

**Pull request merged**